spacy/pipeline/senter.pyx

```diff
@@ -34,19 +36,35 @@ DEFAULT_SENTER_MODEL = Config().from_str(default_model_config)["model"]
 @Language.factory(
     "senter",
     assigns=["token.is_sent_start"],
-    default_config={"model": DEFAULT_SENTER_MODEL},
+    default_config={"model": DEFAULT_SENTER_MODEL, "scorer": {"@scorers": "spacy.senter_scorer.v1"}},
     default_score_weights={"sents_f": 1.0, "sents_p": 0.0, "sents_r": 0.0},
 )
-def make_senter(nlp: Language, name: str, model: Model):
-    return SentenceRecognizer(nlp.vocab, model, name)
+def make_senter(nlp: Language, name: str, model: Model, scorer: Optional[Callable]):
+    return SentenceRecognizer(nlp.vocab, model, name, scorer=scorer)
+
+
+def senter_score(examples, **kwargs):
+    validate_examples(examples, "SentenceRecognizer.score")
```

Comment:
The reference to `SentenceRecognizer` should probably be generalized to use the component's (class) name instead, as this function is also used by the Sentencizer. I guess the same is true for all other registered scoring components, as they can also be used in different components. We could also just call `validate_examples` with `method="score"` and avoid the whole naming altogether...

Reply:
I think `"score"` might be okay. Or maybe we could skip the `validate_examples` in all the individual scoring functions and just leave it `Language.evaluate` instead? I think we can live with not catching this error in advance in the individual `.score` methods? I haven't done any timing tests, but it's a lot of repetitive checking.

Reply:
I've actually been wondering whether we truly need this, yea. I wouldn't mind removing it, it would certainly make things cleaner.